core/src/builder.ts

```diff
@@ -34,8 +30,12 @@ const createState = <Model>({
 
   return {
     get: () => state,
-    merge: (update: Partial<Model>) => {
-      state = { ...state, ...update };
+    merge: (update: Partial<Model>, overwrite = true) => {
```

Comment:
Changes in this object are required so we can initialize the state with the generator result after the builder is actually built.

We need a state in the builder so chained calls to its properties can be registered but we cannot set the initial state at builder instantiation time as we don't know yet which version of the builder we're going to use at that point (this is because the compatibility mode).

This solution allows to update the state with the generator result later during the actual build call.